hazelcast/src/main/java/com/hazelcast/client/impl/ClientHeartbeatMonitor.java

```diff
@@ -76,7 +76,9 @@ public void run() {
     }
 
     private void monitor(String memberUuid, ClientEndpointImpl clientEndpoint) {
-        if (clientEndpoint.isFirstConnection() && ClientType.CPP.equals(clientEndpoint.getClientType())) {
+        // C++ client sends heartbeat over its non-owner connections
+        // For other client types, disregard non-owner connections for heartbeat monitoring purposes
+        if (clientEndpoint.isOwnerConnection() == ClientType.CPP.equals(clientEndpoint.getClientType())) {
```

Comment:
Do not enforce heartbeat monitor:
- on owner connection of C++ client
- on non-owner connections of other clients